snapcraft/internal/project_loader/_parts_config.py

```diff
@@ -149,16 +149,47 @@ def validate(self, part_names):
                     'The part named {!r} is not defined in '
                     '{!r}'.format(part_name, self._snapcraft_yaml))
 
-    def load_plugin(self, part_name, plugin_name, part_properties):
-        part = pluginhandler.load_plugin(
-            part_name,
+    def load_part(self, part_name, plugin_name, part_properties):
+        # Some legacy parts can have a '/' in them to separate the main project
+        # part with the subparts. This is rather unfortunate as it affects the
+        # the layout of parts inside the parts directory causing collisions
+        # between the main project part and its subparts.
+        part_name = part_name.replace('/', '\N{BIG SOLIDUS}')
+
+        plugin = pluginhandler.load_plugin(
             plugin_name=plugin_name,
-            part_properties=part_properties,
+            part_name=part_name,
+            properties=part_properties,
             project_options=self._project_options,
             part_schema=self._validator.part_schema,
             definitions_schema=self._validator.definitions_schema)
 
-        self.build_tools += part.code.build_packages
+        part_schema = {}
+        definitions_schema = {}
+        project_options = snapcraft.ProjectOptions()
```

Comment:
why not `self._project_options`?

Reply:
It actually is, [below](https://github.com/snapcore/snapcraft/pull/1507/files/7cdd84e2d81c85871c990a0d6f7c60b8a7bad5b7#diff-e13dcb301fd4acd6302bee50b3091b03R178). This is keeping the same checks as was present before the refactor. I wasn't sure why there were there, but they seemed relatively harmless and removing them made me nervous. We can try though, and see what the integration tests think.

Reply:
It should just work, this is probably to avoid unit test changes which I don't think would fail today. Please give it a try and remove that if below

Reply:
Yeah I was able to get rid of the other checks as well.